### PR TITLE
perf(aio): serve gzipped content from the preview server

### LINKED ISSUE #16699: the preview server doesn't serve content gzip-ed
screen shot 2017-05-10 at 8 13 43 am
screen shot 2017-05-10 at 8 13 29 am

### PR DESCRIPTION
**Pull request by gkalpak**
Fixes #16699

### CODE REVIEW

**Comment by petebacondarwin** [comment]
I think that the commit type is really `build` not `perf`, since this is only for preview.

File: aio/aio-builds-setup/dockerbuild/nginx/aio-builds.conf [hunk 1]
```diff
@@ -27,6 +27,10 @@ server {
   disable_symlinks on from=$document_root;
   index            index.html;
 
+  gzip            on;
+  gzip_comp_level 7;
```
Comment (petebacondarwin) [inline aio/aio-builds-setup/dockerbuild/nginx/aio-builds.conf]: why did you choose level 7 (out of 9) - was it an arbitrary choice?

**Comment by petebacondarwin** [comment]
Is our nginx sitting behind a proxy at all? If so then we would need to also turn on `gzip_proxied`.

**Review by petebacondarwin — APPROVED** [review]
A few questions but seems to look good to me.
I am worried that despite the gzipping, upping the min-score for PWA to 95 will continue to create flakes in building, since the network latency often causes the current value to drop as low as 91.

Is it worth turning off that check somehow?

**Comment by petebacondarwin** [comment]
Just realised that we need root approval before merging this one.

File: aio/aio-builds-setup/dockerbuild/nginx/aio-builds.conf (hunk 1 above)
Comment (IgorMinar) [inline aio/aio-builds-setup/dockerbuild/nginx/aio-builds.conf]: he compared to the results from firebase

**Review by IgorMinar — APPROVED** [review]

**Comment by gkalpak** [comment]
Regarding the questions:

> Is our nginx sitting behind a proxy at all?

No, it is not (afaik :smiley:)

##
> I am worried that despite the gzipping, upping the min-score for PWA to 95 will continue to create flakes in building, since the network latency often causes the current value to drop as low as 91.

We do PWA score checks in two places: 

1. In `test.sh`, testing against our locally built app, because we are not able to test the deployed version until after it has been deployed. This is a sort of a smoke test, to make sure haven't done something very wrong (like disabling the SW or something).

2. In `deploy.sh`, testing against the deployed app.

The first one currently has a min score of 90% and typically scores ~93%. Being run against a local server, this run is immune to network latency, so the score is pretty stable (I don't remember it ever failing). It has a lower min score, because of the lack of gzipping.
This PR does not affect that test's min score.

The second test used to score 93% most of the time (with some flakes every now and then). After enablign gzipping, the score on my machine improved by ~6% (this is indeed what happened on Travis - currently scoring 99%). So, with 95% we have a pretty safe margin (in fact we might want to increase the min score), unless we do something to really hurt PWA-ability :smiley:

##
> I think that the commit type is really `build` not perf, since this is only for preview.

I was wondering about that, but now I realize `perf` will show up in the changlog which doesn't sound right. So, changing it to `build`.

**Pull request merged**